kr8s/tests/test_api.py

```diff
@@ -414,3 +416,27 @@ async def test_dynamic_classes(kind, ensure_gc):
 async def test_get_dynamic_plurals(kind, ensure_gc):
     api = await kr8s.asyncio.api()
     assert isinstance([resource async for resource in api.get(kind)], list)
+
+
+async def test_two_pods(example_pod_spec, ns):
+    pod1 = await Pod(example_pod_spec)
+
+    example_pod_spec_2 = deepcopy(example_pod_spec)
+    example_pod_spec_2["metadata"]["name"] = "example-" + uuid.uuid4().hex[:10]
+    pod2 = await Pod(example_pod_spec_2)
+
+    pods = [pod1, pod2]
+    for pod in pods:
```

Comment:
This snippet makes me think we might want a nicer API for creating multiple pods. Something that uses an async task group under the hood to concurrently create them. Just a thought for the future.

Reply:
I'm -0.5 on extending the API to more easily create Pods. Even for this simple test, it requires waiting for the pod to be ready. If `kr8s` wants a simple create API, it'll likely have to add more API for a polling interval to check if the pod is ready.

I updated the PR to use task groups in the test, and I do not think it's too complex for a user to do.

Reply:
I'm not sure. You're using the async API so you have the option of using a task group. The sync API can't do this, but we could do it under the hood for you.

Also if you want to wait for multiple Pods to be ready I would just use the `wait` API. It makes a single API request and subscribes to an even stream, then once we've seen a ready event for all the Pods we continue. It's much more efficient that repeatedly polling for every Pod.

https://docs.kr8s.org/en/stable/autoapi/kr8s/objects/index.html#kr8s.objects.APIObject.wait

I wonder if we want a bulk wait operation where you can wait for many resources too.

Reply:
I think creating and waiting for multiple pods is useful. But it largely depends on how much scope do you see kr8s taking.

Concretely, I can see this API for waiting on multiple pods:

```python
pods = [pod1, pod2, ...]

Pod.create_and_wait(pods, conditions, mode, timeout)
```

Reply:
One of the goals of the project is to match the scope and API of `kubectl`. If you can do it with `kubectl` you should be able to do it with `kr8s`, and the API should feel familiar.

In kubectl you would do something like:

```bash
kubectl create -f manifest_of_many_pods.yaml
kubectl wait --for=condition=Ready -f manifest_of_many_pods.yaml
```

So in `kr8s` I would expect it to be something like

```python
import kr8s

many_pods = kr8s.objects.objects_from_files("manifest_of_many_pods.yaml")
kr8s.create(many_pods)
kr8s.wait(many_pods, "condition=Ready")
```